### PR TITLE
Add docs and seed commands

### PR DESCRIPTION
**Pull request by Redande**

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+then in another terminal:
+
+1. `cd backend`
+2. `npm ci`
+3. `sudo -u postgres psql` to open psql terminal
+4. In psql terminal run `CREATE ROLE prisma WITH SUPERUSER CREATEDB CREATEROLE LOGIN ENCRYPTED PASSWORD 'prisma';`
+5. ctrl+d to exit the psql terminal
+6. `psql -U prisma -h 0.0.0.0 -p 5678 -W`
+7. Enter the password that was created in step 4 ("prisma")
```
Comment (mipyykko) [inline docs/installation.md:28]: Is this missing some step (most likely: `CREATE DATABASE ...`) after this, or why are we first creating the `prisma` role, exiting the PSQL terminal and then re-entering it again? The commands below are not PSQL commands, and there are two steps numbered 7. (Markdown will probably create correct numbering, but anyway.)

File: docs/database.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+## Miscellaneous stuff
+
+### Check pod status
+
+`kubectl get pods`
+
+### Check logs
+
+`stern <pod cluster prefix>`, prefix is e.g. moocfi-backend
```
Comment (mipyykko) [inline docs/database.md:75]: `stern` needs to be installed separately

Other useful kube commands:

* `kubectl logs <pod_name>` - get logs for a single pod
   - where `pod_name` is the full name with suffix like `fetch-avoin-links-27777615-xyzzy` 
   - what `stern` does is basically `kubectl logs <prefix>*` 
* `kubectl get jobs`
* `kubectl get cronjobs`
* `kubectl describe <resource> <resource_name>` - describe resource
   - where resource name like `job`, `pod` - get full list with `kubectl api-resources`
   - useful for debugging why a pod or job might have failed, for example `kubectl describe job sync-tmc-users-27780760`
    - will help finding the logs for a recent job/cronjob - you can find the name for the pod that was created with describe, although the pods will be deleted after a set time
* `kubectl create job --from=cronjob/<cronjob_name> <job_name>` - run a cronjob manually
    - come up with a new unique job name; you can delete the job later when it's finished with `kubectl delete job <job name>`

File: docs/database.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+2. give a name to the source
+3.
+* `localhost` in host field,
+* `3306` in port field,
+* `postgres` in user field,
+* db_secret from earlier to the password field,
+* `postgres` in database field
+4. in Schemas-tab, check which dbs you want (probably certificates, exams, moocfi, quizzes1, 2 and 3, staging, maybe others based on usage)
+5. refresh the data source in the database explorer to see the databases
+6. for moocfi -db, set schema in console to `moocfi$production` (this must be done every time you use datagrip)
```
Comment (mipyykko) [inline docs/database.md:46]: Can be set to remember this on console/SQL file basis; needs to be set again for each new window, though: https://www.jetbrains.com/help/datagrip/schemas.html#save-a-search-path-between-ide-restarts

File: docs/database.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+## local
+
+1. `psql -h localhost -p 5678 -U prisma -W`
+2. enter password `prisma`
+3. run `set search_path to "default$default";`
+
+## production
+
+1. in a terminal window run
+`cloud_sql_proxy -instances=moocfi:europe-north1:quizzes-backend=tcp:0.0.0.0:3306`
```
Comment (mipyykko) [inline docs/database.md:15]: `cloud_sql_proxy` has to be installed separately; an alternative to running the proxy every time is to add your IP range to the Google Cloud Console (SQL -> Connections -> Authorized networks)

File: docs/database.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+* `localhost` in host field,
+* `3306` in port field,
+* `postgres` in user field,
+* db_secret from earlier to the password field,
+* `postgres` in database field
+4. in Schemas-tab, check which dbs you want (probably certificates, exams, moocfi, quizzes1, 2 and 3, staging, maybe others based on usage)
+5. refresh the data source in the database explorer to see the databases
+6. for moocfi -db, set schema in console to `moocfi$production` (this must be done every time you use datagrip)
+
+now you can run sql commands in console, test for example `select * from course;` and you should see courses listed in output window.
```
Comment (mipyykko) [inline docs/database.md]: One useful hint to be added here is to tell Datagrip to open a new result window for each query: https://www.jetbrains.com/help/datagrip/viewing-query-results.html#open-a-new-tab-for-each-query

File: docs/installation.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+for backend:
+
+1. `cd backend`
+2. `docker-compose up`
+
+then in another terminal:
+
+1. `cd backend`
+2. `npm ci`
+3. `sudo -u postgres psql` to open psql terminal
```
Comment (mipyykko) [inline docs/installation.md:24]: Does this connect to the right PSQL instance?

File: docs/database.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+## production
+
+1. in a terminal window run
+`cloud_sql_proxy -instances=moocfi:europe-north1:quizzes-backend=tcp:0.0.0.0:3306`
+to proxy production to localhost port 3306
+
+when connecting first time, run the following to get db_name and db_secret
+
+1. `kubectl get secrets`
```
Comment (mipyykko) [inline docs/database.md:26]: `kubectl` needs to be separately installed; also needs credentials set to access the cluster: https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl - `gcloud container clusters get-credentials <CLUSTER_NAME>`

Change to correct namespace using `kubectl ns moocfi`

File: backend/package.json [hunk 8]
```diff
@@ -28,8 +28,10 @@
     "generate:prisma": "prisma generate",
     "kafka-status": "node ./dist/bin/kafkaStatus.js",
     "seed": "node ./dist/bin/seed.js",
+    "seedAll": "npm run seed && npm run seedPoints && npm run seedExercises && npm run seedCompletions",
```
Comment (mipyykko) [inline backend/package.json:31]: Might want to add these to the "first time run"; also mention that you need to do `npm run build` for these to work (or does it create dist with `npm run dev`, can't remember)

**Review by mipyykko — CHANGES_REQUESTED** [review]

File: backend/package.json (hunk 8 above)
Comment (Redande) [reply backend/package.json:31]: Good point, added

File: docs/database.md (hunk 4 above)
Comment (Redande) [reply docs/database.md:15]: Another good point, added installation instructions for `cloud_sql_proxy` and mentioned the alternative

File: docs/database.md (hunk 7 above)
Comment (Redande) [reply docs/database.md:26]: Added :+1:

File: docs/database.md (hunk 3 above)
Comment (Redande) [reply docs/database.md:46]: Added to guide :+1:

File: docs/installation.md (hunk 6 above)
Comment (Redande) [reply docs/installation.md:24]: Yes

File: docs/installation.md (hunk 1 above)
Comment (Redande) [reply docs/installation.md:28]: Yes, this was missing the step `create schema "default$default";`
We first enter the terminal with user `psql` to create the user `prisma`, then we re-enter with the newly created user `prisma` to create the schema

Fixed the numbering

**Review by mipyykko — APPROVED** [review]

**Pull request merged**